GRDB/QueryInterface/Schema/TableDefinition.swift

```diff
@@ -806,6 +820,32 @@ public final class ColumnDefinition {
         return self
     }
     
+    /// Adds a generated column constraint on the column.
+    ///
+    ///     try db.create(table: "player") { t in
+    ///         t.column("id", .integer).primaryKey()
+    ///         t.column("score", .integer).notNull()
+    ///         t.column("bonus", .integer).notNull()
+    ///         t.column("totalScore", .integer).generatedAs(sql: "score + bonus", .stored)
+    ///     }
+    ///
+    /// See https://sqlite.org/gencol.html. Note particularly the limitations of generated
+    /// columns, e.g. they may not have a default value.
+    ///
+    /// - parameters:
+    ///     - sql: An SQL expression.
+    ///     - qualification: The generated column's qualification.
+    /// - returns: Self so that you can further refine the column definition.
+    @discardableResult
+    public func generatedAs(sql: String, _ qualification: GeneratedColumnQualification) -> Self
```

Comment:
This method must be wrapped in `#if GRDBCUSTOMSQLITE ... #endif`, as other apis that are only available in custom builds. We do not want to expose apis that fail at runtime on some deployment targets.

When iOS/macOS/etc ship with support for generated columns, we'll add a mix of `@available` and `#if GRDBCUSTOMSQLITE`. We're not there yet.

Reply:
I'd like to find a way to conditionally compile this API method for a minimum version of SQLite, irregardless of whether the SQLite build is custom or not. Sadly, I have not found a way. I'm assuming you haven't either.

Reply:
No, there is no way I'm aware of. We have to duplicate declarations: one set is guarded by `@available`, the other set is guarded by `#if`. For small sets of declarations, this is kind of OK. For large sets, such as FTS5 support, this is a total blocker (see #819).